api/src/test/java/org/apache/iceberg/geospatial/TestGeospatialPredicateEvaluators.java

```diff
@@ -0,0 +1,489 @@
+    BoundingBox box1 = new BoundingBox(min1, max1);
+
+    GeospatialBound min2 = GeospatialBound.createXY(2.0, 2.0);
+    GeospatialBound max2 = GeospatialBound.createXY(5.0, 5.0);
+    BoundingBox box2 = new BoundingBox(min2, max2);
+
+    assertThat(evaluator.intersects(box1, box2)).isTrue();
+    assertThat(evaluator.intersects(box2, box1)).isTrue();
+  }
+
+  @Test
+  public void testGeometryBoxesWithZCoordinate() {
+    Type geometryType = Types.GeometryType.crs84();
+    GeospatialPredicateEvaluators.GeospatialPredicateEvaluator evaluator =
+        GeospatialPredicateEvaluators.create(geometryType);
+
+    // Two boxes with Z coordinates that overlap in X and Y but not in Z
+    GeospatialBound min1 = GeospatialBound.createXYZ(0.0, 0.0, 0.0);
+    GeospatialBound max1 = GeospatialBound.createXYZ(2.0, 2.0, 1.0);
+    BoundingBox box1 = new BoundingBox(min1, max1);
+
+    GeospatialBound min2 = GeospatialBound.createXYZ(1.0, 1.0, 2.0);
+    GeospatialBound max2 = GeospatialBound.createXYZ(3.0, 3.0, 3.0);
+    BoundingBox box2 = new BoundingBox(min2, max2);
+
+    GeospatialBound min3 = GeospatialBound.createXYZ(1.0, 1.0, 1.0);
+    GeospatialBound max3 = GeospatialBound.createXYZ(3.0, 3.0, 3.0);
+    BoundingBox box3 = new BoundingBox(min3, max3);
+
+    assertThat(evaluator.intersects(box1, box2)).isFalse();
+    assertThat(evaluator.intersects(box2, box1)).isFalse();
+    assertThat(evaluator.intersects(box1, box3)).isTrue();
+    assertThat(evaluator.intersects(box3, box1)).isTrue();
+    assertThat(evaluator.intersects(box2, box3)).isTrue();
+    assertThat(evaluator.intersects(box3, box2)).isTrue();
+  }
+
+  @Test
+  public void testGeometryBoxesWithMCoordinate() {
+    Type geometryType = Types.GeometryType.crs84();
+    GeospatialPredicateEvaluators.GeospatialPredicateEvaluator evaluator =
+        GeospatialPredicateEvaluators.create(geometryType);
+
+    // Two boxes with M coordinates that overlap in X and Y but not in M
+    GeospatialBound min1 = GeospatialBound.createXYM(0.0, 0.0, 0.0);
+    GeospatialBound max1 = GeospatialBound.createXYM(2.0, 2.0, 1.0);
+    BoundingBox box1 = new BoundingBox(min1, max1);
+
+    GeospatialBound min2 = GeospatialBound.createXYM(1.0, 1.0, 2.0);
+    GeospatialBound max2 = GeospatialBound.createXYM(3.0, 3.0, 3.0);
+    BoundingBox box2 = new BoundingBox(min2, max2);
+
+    GeospatialBound min3 = GeospatialBound.createXYM(1.0, 1.0, 1.0);
+    GeospatialBound max3 = GeospatialBound.createXYM(3.0, 3.0, 3.0);
+    BoundingBox box3 = new BoundingBox(min3, max3);
+
+    assertThat(evaluator.intersects(box1, box2)).isFalse();
+    assertThat(evaluator.intersects(box2, box1)).isFalse();
+    assertThat(evaluator.intersects(box1, box3)).isTrue();
+    assertThat(evaluator.intersects(box3, box1)).isTrue();
+    assertThat(evaluator.intersects(box2, box3)).isTrue();
+    assertThat(evaluator.intersects(box3, box2)).isTrue();
+  }
+
+  @Test
+  public void testGeometryBoxesWithEmptyXRange() {
```

Comment:
what does `EmptyXRange` mean? 

Is this trying to test X wraparound for geometry, which was not allowed with latest? It makes me think that `GeometryEvaluator` should throw an exception when the bounding box has X wraparound, similar to the `GeographyEvaluator#validateBoundingBox` method

Reply:
Right now, the bounding box validation is done in the predicate evaluator. maybe the BoundingBox validations should happen in the construction of BoundingBox object. 

One possible approach is using two public factory methods to create BoundingBox objects.
```
public static BoundingBox forGeometry() {}
public static BoundingBox forGeograph() {}
```

Those factory methods can add specific type related validations.